scripts.js

```diff
@@ -225,6 +222,169 @@ window.adobeid = {
+    }
+  ];
+  
+  // Add Region Dropdown Container before Feds Footer
+  const fedsFooter = document.querySelector('#feds-footer');
+  if (fedsFooter) {
+    const regionDropdownContainer = document.createElement('div');
+    regionDropdownContainer.classList.add('region-dropdown');
+    regionDropdownContainer.innerHTML = `<ul class="region-dropdown-list"></ul>`;
+    fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
+  }
+  
+  // get actual selected Region
+  const getSelectedRegion = () => {
+    const regionPage = regionsNameList.find(r => r.home === location.pathname);
+    // check if Region name matches the blog.language example 'en' in order to show the Region Name in the buttom
+    const nonRegionPage = regionsNameList.find(r => r.lang === window.blog.language);
+    let regionLang;
+    let regionName;
+    if (!regionPage) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use blog.language
+      regionLang = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+      // in order to show the Region name either from SessionStorage language or blog.language in our region button
+      if (regionLang !== window.blog.language) {
+        // if the selected Region lang do not match the blog.language we get the Region name based on the sessionStorage saved Language Value
+        const storedLanguage = regionsNameList.find(r => r.lang === regionLang);
+        regionName = storedLanguage.name;
+      } else {
+        // else we get the Region name base on the blog.language and after have find the same language in our RegionListName
+        if (nonRegionPage !== undefined) {
+            regionName = nonRegionPage.name;
+        }       
+      }
+    } else {
+      regionLang = regionPage.lang;
+      regionName = regionPage.name;
+      // sessionStorage will be used only if current Region lang is not same as blog.lang as en_apac or en_uk 
+      // where blog.lang will still being en
+      if (regionPage.lang !== window.blog.language) {
+        sessionStorage.setItem('blog-selected-language', regionPage.lang);
+      }
+    } 
+    return {regionLang, regionName};
+  }
+
+  const dropdownRegionList = document.querySelector('.region-dropdown-list');
+  const {regionLang, regionName} = getSelectedRegion(); 
+
+  // Change Region name value from Feds Region Picker Button adding the actual Region Name
+  const FEDSregionPickerText = document.querySelector('.feds-regionPicker-text');
+  if (FEDSregionPickerText && regionName !== undefined) {
+      FEDSregionPickerText.innerText = regionName;
+  }
+
+  // Automatically build the dropdown based on Region List
+  if (dropdownRegionList) {
+    for (const {lang, name, home} of regionsNameList) {
+      dropdownRegionList.insertAdjacentHTML('afterbegin', `<li><a class="region-dropdown-picker" href="${window.location.origin + home}" title="${name}" data-lang="${lang}">${name}</a></li>`);
```

Comment:
You probably don't need the `window.location.origin`
```suggestion
      dropdownRegionList.insertAdjacentHTML('afterbegin', `<li><a class="region-dropdown-picker" href="${home}" title="${name}" data-lang="${lang}">${name}</a></li>`);
```

Reply:
Updated this one